Fixes /welcomebot preview teamname command not working.

maisnamrajusingh opened the pull request:
#### Summary
- Fixes issue of `/welcomebot preview teamname` command not working.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-welcomebot/issues/71

mattermod commented:
Hello @maisnamrajusingh,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

hanzei commented on server/command.go:
```diff
@@ -88,7 +88,7 @@ func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs)
 	for _, message := range p.getWelcomeMessages() {
 		if message.TeamName == teamName {
-			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
+			if err := p.previewWelcomeMessage(args, *message); err != nil {
```
While I think the behavior change is reasonable, this makes `teamName` not longer needed. Would you please remove it, also update the slash command auto complete and `README.md`?

Thanks!

maisnamrajusingh replied:
Sure, Will do.

maisnamraju replied:
@hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

maisnamrajusingh replied:
@hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

hanzei replied:
Would you mind including the docs and autocomplete changes in this PR? That makes it easier to review if a PR is complete-

jasonblais replied:
cc @maisnamrajusingh on the above note from hanzei

maisnamrajusingh replied:
got it guys.

maisnamrajusingh replied:
@hanzei  I think, there  is a small confusion here. There is no need to change the docs in this case as we are still using the `teamName` to compare the config values that we get from the plugin config. Please check line no 90 on the same file.

hanzei replied:
Could you please elaborate why `teamName` is check, if `previewWelcomeMessage` uses `args.TeamId` anyway?

maisnamrajusingh replied:
@hanzei  the way the command works is `/welcomebot preview <TeamName>` so we need the team name to check which messages in the config relate to the particular team in the command.

hanzei replied:
The team name passed to the command is not longer coupled with the `args.TeamId` used to post the preview. Hence a user could use `/welcomebot preview SomeOtherTeamName` and still get a preview for the current team.

maisnamrajusingh replied:
@hanzei  sorry I thought this pr was merged, let me look into this today and get back to you

jasonblais replied:
@maisnamrajusingh Any updates on this PR, or an ETA when it's ready for a review?

maisnamrajusingh replied:
@hanzei  sorry, it took so long to respond to this. If you look at the code, you can see that the teamName is still used to make the comparison to check if the message belongs to that team. Or did I misunderstand your comment?

```
func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) {
	found := false
	for _, message := range p.getWelcomeMessages() {
		if message.TeamName == teamName {
			if err := p.previewWelcomeMessage(args, *message); err != nil {
				p.postCommandResponse(args, "error occurred while processing greeting for team `%s`: `%s`", teamName, err)
				return
			}

			found = true
		}
	}

	if !found {
		p.postCommandResponse(args, "team `%s` has not been found", teamName)
	}
}
```

jasonblais replied:
cc @hanzei quick reminder on above

hanzei replied:
The current behavior on `master` is: The user has to pass a team name, which is used to post the message.

With your PR the behavior is: The user has to pass a team name, which is used to find the welcome message, but the template itself is executed within the context of the current team.

Either the behavior should stay the same as in `master` or the team name should not longer be required and the current team is used.

Does that make things more clear?

maisnamraju replied:
@hanzei makes sense. will update the logic and update accordingly.

hanzei reviewed (changes requested):
The behavior change is fine, but it needs to be carried through whole code and the docs.

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @jfrerich @emilyacook

maisnamrajusingh commented:
@hanzei  I over thought what the problem was. The issue actually was that the user had `Cudy` as the team name and in the config whereas the `GetTeamByName` was expecting the values in lowercase. I reverted back my previous changes and added a string conversion to lower case when we fetch the team by its name

maisnamrajusingh commented:
@hanzei  can you take a look at this and see if it needs anything else from my side ?

hanzei commented on server/welcomebot.go:
```diff
@@ -57,7 +57,7 @@ func (p *Plugin) getSiteURL() string {
 }
 
-func (p *Plugin) newSampleMessageTemplate(teamName, userID string) (*MessageTemplate, error) {
+func (p *Plugin) newSampleMessageTemplate(teamName string, userID string) (*MessageTemplate, error) {
```
Nit pick: Unrelated change

hanzei reviewed (approved):
Looks good to me :+1:

hanzei commented:
@jfrerich Please take a look

jfrerich reviewed (approved):
LGTM!

hanzei commented:
Deferring QA to release testing
